packages/Webkul/Velocity/webpack.mix.js

```diff
@@ -1,40 +1,79 @@
+const publicPath     = mix.inProduction() ? prodPublicPath : devPublicPath;
 
-if (mix.inProduction()) {
-    publicPath = 'publishable/assets';
-}
+console.log(colors.bold.blue(`Assets will be published in: ${publicPath}`));
 
-mix.setPublicPath(publicPath).mergeManifest();
-mix.disableNotifications();
+const assetsPath = path.join(__dirname, 'src', 'Resources', 'assets');
+const jsPath     = path.join(assetsPath, 'js');
+const imagesPath = path.join(assetsPath, 'images');
 
 mix
-    .js(__dirname + '/src/Resources/assets/js/app-core.js', 'js/velocity-core.js')
-    .js(
-        __dirname + '/src/Resources/assets/js/app.js',
-        'js/velocity.js'
+    .setPublicPath(publicPath)
+
+    .js(path.join(jsPath, 'app-core.js'), 'js/velocity-core.js')
+    .js(path.join(jsPath, 'app.js'), 'js/velocity.js')
+    .vue()
+
+    .alias({
+               '@Components': path.join(jsPath, 'UI', 'components')
+           })
+    .extract({
+                 to: `/js/components.js`,
+                 test(mod){ return /(component|style|loader|node)/.test(mod.nameForCondition()) }
+             }
     )
 
-    .copy(__dirname + '/src/Resources/assets/images', publicPath + '/images')
+    .copy(imagesPath, path.join(publicPath, 'images'))
 
     .sass(
-        __dirname + '/src/Resources/assets/sass/admin.scss',
-        __dirname + '/' + publicPath + '/css/velocity-admin.css'
+        path.join(assetsPath, 'sass', 'admin.scss'),
+        path.join(__dirname, publicPath, 'css', 'velocity-admin.css'),
     )
     .sass(
-        __dirname + '/src/Resources/assets/sass/app.scss',
-        __dirname + '/' + publicPath + '/css/velocity.css',
+        path.join(assetsPath, 'sass', 'app.scss'),
+        path.join(__dirname, publicPath, 'css', 'velocity.css'),
         {
-            includePaths: ['node_modules/bootstrap-sass/assets/stylesheets/']
+            implementation: require('node-sass'),
+            sassOptions   : {
+                includePaths: [ 'node_modules/bootstrap-sass/assets/stylesheets/' ]
+            }
         }
     )
 
+    .clean({
+               // enable `dry` before adding new paths:
+               // dry: true,
+               cleanOnceBeforeBuildPatterns: [
+                   'js/**/*',
+                   'css/velocity.css',
+                   'css/velocity-admin.css',
+                   'mix-manifest.json',
+               ]
+           })
+
     .options({
-        processCssUrls: false
-    });
+                 processCssUrls: false,
+                 clearConsole  : mix.inProduction()
+             })
+
+    .disableNotifications()
+    .mergeManifest();
+
+    // https://laravel-mix.com/docs/6.0/faq#can-i-autoload-modules-with-mix-and-webpack
+    // https://laravel-mix.com/docs/6.0/autoloading
+    /*.autoload({
+                  vue   : [ 'Vue', 'window.Vue' ],
+                  jquery: [ '$', 'window.jQuery', 'jQuery', 'window.$', 'jquery', 'window.jquery' ]
+              })
+    */
```

Comment:
Please clean up this.

Reply:
Just to clarify why I left this code: it's supposed to load automatically jquery or Vue, once any of the listed keywords is found by webpack inside the processed Js files. 
This way is not needed to import Vue or jquery every time, as webpack will take care of it. 

I'm going to remove it now, but it may be required in the future, if you decide to process every js in the project, with webpack.